### DISCUSSION TITLE
Custom event handling

### DISCUSSION THREAD

**Discussion by joebillman** [pixijs/pixijs#9625] [Q&A]
Hi everyone, I'm somewhat new to Pixi. I got my start with ActionScript back in the day, but after the death of Flash have been doing a lot of work on the web using CreateJS/EaselJS. I'm really liking what I'm seeing with Pixi. I have a question regarding custom events. I need to be able to dispatch and event from one object and have another object listen to it and react according to some data included with the event. What is the preferred way to do that in PixiJS 7?

**Comment by SignDawn** [comment]
eventbus ?

**Reply by joebillman** [reply]
I've seen that there a quite a few examples for the EventEmitter online, but I also see that there is an EventSystem class. The EventEmitter looks to be a 3rd party resource. I don't see many examples for the EventSystem. Is that the preferred way or is the EventEmitter?

**Reply by miltoncandelero** [reply]
With pixi 7.3 RC it was fixed that any object can emit a custom event just like flash, however the events don't bubble up the display list like flash did.

Say, I emit "hit" from my character sprite, I can't listen for hit on the stage, I need to listen on that sprite

**Reply by joebillman** [reply]
Thank you that is good to know!

**Comment by RecursiveVoid** [comment]
I mostly use this for my projects;  https://www.npmjs.com/package/eventemitter3
